### PR TITLE
Add objects sub-group

### PR DESCRIPTION
**Pull request by sfc-gh-jsikorski**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
* Refactored `snow stage` and `snow warehouse` to the new `snow object` subgroup
* Changed `snow warehouse status` to `snow object show warehouse`

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 * `snow snowpark registry` was replaced with `snow registry` command.
 * Removed `snow streamlit create` command. Streamlit can be deployd using `snow streamlit deploy`
 * `snow connection test` now outputs all connection details (except for the password), along with connection status
+* Moved `snow stage` and `snow warehouse` from top-level to `snow object` subgroup
```
Comment (sfc-gh-turbaszek) [inline RELEASE-NOTES.md:18]: We agreed to remove `snow warehouse status` (the only wh command) in favour of `snow object show warehouses`

Comment (sfc-gh-jsikorski) [reply RELEASE-NOTES.md:18]: Ok, i was basing on table in https://docs.google.com/document/d/1FLkKWWsKLWVeJ4n0TWARi6voqinpeu54Tf85v1kY3yw/edit#heading=h.67voytek1ow7

Comment (sfc-gh-jsikorski) [reply RELEASE-NOTES.md:18]: BTW. Should i add `snow object drop` etc. as part of this PR?

File: src/snowcli/cli/object/__init__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import typer
+
+from snowcli.cli.common.flags import DEFAULT_CONTEXT_SETTINGS
+from snowcli.cli.object.stage.commands import app as stage_app
+from snowcli.cli.object.show.commands import app as warehouse_app
```
Comment (sfc-gh-astus) [inline src/snowcli/cli/object/__init__.py:5]: `as show_app`

File: src/snowcli/cli/snowpark/__init__.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 app = typer.Typer(
     name="snowpark",
     context_settings=DEFAULT_CONTEXT_SETTINGS,
-    help="Manages functions, procedures, and Snowpark objects.",
+    help="Manages functions, procedures, and Snowpark object.",
```
Comment (sfc-gh-astus) [inline src/snowcli/cli/snowpark/__init__.py:16]: `functions`, `procedures`, I would keep `objects`

File: tests_integration/test_warehouse.py [hunk 4]
```diff
@@ -4,9 +4,11 @@
 from tests_integration.snowflake_connector import snowflake_session
 
 
-@pytest.mark.integration
+# @pytest.mark.integration TODO: remove hash
```
[the remark this answers is not in the document]
Comment (sfc-gh-jsikorski) [reply tests_integration/test_warehouse.py]: Done

File: src/snowcli/cli/snowpark/__init__.py (hunk 3 above)
Comment (sfc-gh-jsikorski) [reply src/snowcli/cli/snowpark/__init__.py]: Done

File: src/snowcli/cli/object/__init__.py (hunk 2 above)
Comment (sfc-gh-jsikorski) [reply src/snowcli/cli/object/__init__.py]: Done

File: tests_integration/test_package.py [hunk 5]
```diff
@@ -18,7 +18,7 @@
 class TestPackage:
     STAGE_NAME = "PACKAGE_TEST"
 
-    @pytest.mark.integration
+    # @pytest.mark.integration
```
Comment (sfc-gh-astus) [inline tests_integration/test_package.py:21]: Uncomment test 😄

File: tests_integration/test_package.py [hunk 6]
```diff
@@ -47,7 +47,7 @@ def test_package_upload(self, runner, snowflake_session, test_database):
 
         snowflake_session.execute_string(f"DROP STAGE IF EXISTS {self.STAGE_NAME};")
 
-    @pytest.mark.integration
+    # @pytest.mark.integration
```
Comment (sfc-gh-astus) [inline tests_integration/test_package.py:50]: And here

File: RELEASE-NOTES.md (hunk 1 above)
Comment (sfc-gh-jsikorski) [reply RELEASE-NOTES.md]: Done

File: tests_integration/test_package.py (hunk 5 above)
Comment (sfc-gh-jsikorski) [reply tests_integration/test_package.py]: Done

File: tests_integration/test_package.py (hunk 6 above)
Comment (sfc-gh-jsikorski) [reply tests_integration/test_package.py]: Done

**Review by sfc-gh-astus — APPROVED** [review]

File: src/snowcli/cli/object/__init__.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import typer
+
+from snowcli.cli.common.flags import DEFAULT_CONTEXT_SETTINGS
+from snowcli.cli.object.stage.commands import app as stage_app
+from snowcli.cli.object.show.commands import app as show_app
+
+app = typer.Typer(
+    name="object",
+    context_settings=DEFAULT_CONTEXT_SETTINGS,
+    help="Manages Snowflake objects, like warehouses and stages",
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/cli/object/__init__.py:10]:
```suggestion
    help="Manages Snowflake objects like warehouses and stages.",
```

File: src/snowcli/cli/object/show/commands.py [hunk 8]
```diff
@@ -4,23 +4,23 @@
...
-from snowcli.cli.warehouse.manager import WarehouseManager
+from snowcli.cli.object.show.manager import ShowManager
 from snowcli.output.decorators import with_output
 from snowcli.output.types import QueryResult
 
 app = typer.Typer(
-    name="warehouse",
+    name="show",
     context_settings=DEFAULT_CONTEXT_SETTINGS,
     help="Manages warehouses.",
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/cli/object/show/commands.py:14]:
```suggestion
    help="List existing objects of given type",
```
Also we should make it a command with single argument `[TYPE]` and `--like` flag.

File: src/snowcli/cli/object/show/commands.py [hunk 9]
```diff
@@ -4,23 +4,23 @@
...
-@app.command("status")
+@app.command("warehouses")
 @with_output
 @global_options_with_connection
 def warehouse_status(**options):
     """
     Shows the status of each warehouse in the configured environment.
     """
-    cursor = WarehouseManager().show()
+    cursor = ShowManager().show()
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/cli/object/show/commands.py:25]: I think we should have more generic manager that will have the following methods:
```
def describe(object_type: str, name:str)

def drop(object_type: str, name:str)

def show(object_type: str)
```

Such commands probably can be incorporated in `SqlExecutionMixin`. WDYT?

File: src/snowcli/cli/object/commands.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+@app.command()
+@with_output
+@global_options_with_connection
+def show(
+    object_type: ObjectType = ObjectArgument,
+    like: str = LikeOption,
+    **options,
+):
+    "Lists all avaiable Snowflake objects of given type"
```
Comment (sfc-gh-astus) [inline src/snowcli/cli/object/commands.py:39]: `available`

File: src/snowcli/cli/object/utils.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
+from enum import Enum
+from typing import Dict
+
+
+class ObjectType(str, Enum):
```
[the remark this answers is not in the document]
Comment (sfc-gh-jsikorski) [reply src/snowcli/cli/object/utils.py]: Done

File: src/snowcli/cli/object/commands.py (hunk 10 above)
Comment (sfc-gh-jsikorski) [reply src/snowcli/cli/object/commands.py]: Done

File: RELEASE-NOTES.md [hunk 12]
```diff
@@ -25,10 +25,15 @@
...
+* Moved `snow stage` from top-level to `snow object` subgroup
+* `snow warehouse status` is now `snow object show warehouse`
+* Introduced `snow object` group with `show`, `describe` and `drop` commands for: compute pools,
+databases, tables, warehouses, functions, procedures, roles, schemas, services, jobs and streamlits
+
 
 ## New additions
 * `--temporary-connection` flag, that allows you to connect, without anything declared in config file
-* Added project definition for Streamlit
+* Added project definition for Streamlitpytest
```
Comment (sfc-gh-astus) [inline RELEASE-NOTES.md:36]: Wrong window case? 😄

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**